Tests/EndToEnd.tests.ts

```diff
@@ -153,6 +153,18 @@ class fakeHttpsServer extends events.EventEmitter {
 }
 
 describe("EndToEnd", () => {
+    var originalEnv = {};
+
+    before(() => {
```

Comment:
E2E tests here rely on some fakeHttpServer serving all requests, now that Azure VM request is in place is making these tests unreliable, we need to refactor this code and use nock to handle all SDK HTTP requests correctly, created a task to take care of that in different PR